specs/trip_dispatch_spec.rb

```diff
@@ -88,5 +86,84 @@
       passenger.must_be_instance_of RideShare::Passenger
       passenger.trips.must_include trip
     end
+  end # end of describe "loader methods"
+
+  describe "available_driver method" do
+    before do
+      @dispatcher = RideShare::TripDispatcher.new
+    end
+
+    it "accurately finds the fist available driver" do
+      first_driver = @dispatcher.available_driver
+
+      first_driver.name.must_equal "Emory Rosenbaum"
+      first_driver.id.must_equal 2
+      first_driver.status.must_equal :AVAILABLE
+    end
+
+    it "accurately finds the next available driver" do
+      trip = @dispatcher.request_trip(21)
+      second_driver = trip.driver
+      second_driver = @dispatcher.available_driver
+
+      second_driver.name.must_equal "Daryl Nitzsche"
+      second_driver.id.must_equal 3
+      second_driver.status.must_equal :AVAILABLE
+    end
+
+    # TODO: add a edge case here
```

Comment:
An example of an edge case would be, what happens if there are no available drivers? This is exactly the sort of situation where software will do something unexpected. You might set up the test as follows:

```ruby
it 'raises an error if no driver is available' do
  @dispatcher.drivers.each do |driver|
    # This may require an attr_writer in Driver
    driver.status = :UNAVAILABLE
  end
  proc {
    @dispatcher.available_driver
  }.must_raise
end
```